crates/fluvio-connector-package/src/secret.rs

```diff
@@ -75,6 +93,19 @@ impl From<Secret> for SecretString {
 
 pub trait SecretStore: Send + Sync {
     fn read(&self, name: &str) -> Result<String>;
+
+    fn resolve_str(&self, input: &str) -> Result<String> {
+        let result = SECRETS_RE
+            .replace_all(input, |caps: &regex::Captures| {
+                let secret_name = caps
+                    .get(1)
+                    .expect("should not fail since SECRETS_RE, contains a capture group")
```

Comment:
Even if it is expected to never happen, I would change it for proper error propagation instead of panicking. This is a third-partly library, they can change the behavior.

Reply:
Resolved with https://github.com/infinyon/fluvio/pull/3138/commits/c65e6ddfcabbed2bfe1a2c0adc8f47de3f933ff7